### PR TITLE
Add ROS2 object detection 2D, face detection and semantic segmentation nodes

### PR DESCRIPTION
**Pull request by tsampazk**
Nodes additions:
- Object Detection 2D - SSD
- Object Detection 2D - Centernet
- Object Detection 2D - Detr
- Object Detection 2D - YOLOv3
- Face Detection Retinaface
- Semantic Segmentation Bisenet
- Added entries in setup.py for nodes

Bridge additions:
- Added to and from ros bounding box list methods
- Added to and from ros boxes methods

Other:
This PR includes minor improvements in yolov3, ssd and centernet learners to stop them re-downloading the trained models if they are previously downloaded.

Finalization:
- [x] Renaming of packages
- [x] Bridge import improvements
- [x] Complete descriptions, license, maintainer, etc. information
- [x] Bridge methods
- [x] Pose Estimation
- [x] Semantic Segmentation Bisenet
- [x] Face Detection Retinaface
- [x] Object Detection 2D - SSD
- [x] Object Detection 2D - Centernet
- [x] Object Detection 2D - Detr
- [x] Object Detection 2D - YOLOv3

### CODE REVIEW

File: projects/opendr_ws_2/src/opendr_perception/package.xml [hunk 1]
```diff
@@ -1,19 +1,19 @@
...
 <?xml-model href="http://download.ros.org/schema/package_format3.xsd" schematypens="http://www.w3.org/2001/XMLSchema"?>
 <package format="3">
-  <name>perception</name>
-  <version>0.0.0</version>
-  <description>TODO: Package description</description>
-  <maintainer email="test@csd.auth.gr">test</maintainer>
-  <license>TODO: License declaration</license>
+  <name>opendr_perception</name>
+  <version>1.0.0</version>
+  <description>penDR ROS2 nodes for the perception package</description>
```
Comment (stefaniapedrazzi) [inline projects/opendr_ws_2/src/opendr_perception/package.xml:6]:
```suggestion
  <description>OpenDR ROS2 nodes for the perception package</description>
```

**Review by stefaniapedrazzi — CHANGES_REQUESTED** [review]
Thank you!
This PR seems fine to me.

**Comment by tsampazk** [comment]
Thanks for the quick review @stefaniapedrazzi, i'll go ahead and squash&merge as soon as the tests complete.

**Comment by tsampazk** [comment]
Hey @jelledouwe, i noticed that you are working on a ROS2 DETR object detection 2D node. In this PR i have already added the object detection 2D DETR node. I am sorry for the miscommunication on my part. Maybe you could take a look at it [here.](https://github.com/opendr-eu/opendr/blob/ros2/projects/opendr_ws_2/src/opendr_perception/opendr_perception/object_detection_2d_detr_node.py) I have made the needed necessary changes for ROS2, as well as changes for internal consistency and consistency with other nodes.

Maybe you could modify that file straight away from a new branch out of ros2 branch if you feel that it needs changes?

**Pull request merged**